shared/common-adapters/markdown.js

```diff
@@ -37,165 +34,59 @@ const codeSnippetBlockStyle = {
-const openToClosePair = {
-  '```': '```',
-  '`': '`',
-  '*': '*',
-  '_': '_',
-  '~': '~',
-  ':': ':',
-}
-
-// We have to escape certain marks when turning them into a regex
-const markToRegex = {
-  '```': '```',
-  '`': '`',
-  '*': '\\*',
-  '_': '_',
-  '~': '~',
-  ':': ':',
-}
-
-const plainStringTag = {Component: Text, props: {type: 'Body', style: {color: undefined}}}
-
 class EmojiIfExists extends PureComponent<void, EmojiProps, void> {
   render () {
-    const emoji = this.props.children && this.props.children.join('')
-    const exists = emojiIndex.emojis.hasOwnProperty(emoji)
-    return exists ? <Emoji {...this.props} /> : <Text {...plainStringTag.props}>:{emoji}:</Text>
+    const emoji = (this.props.children && this.props.children.join('')) || ''
+    const exists = emojiIndex.emojis.hasOwnProperty(emoji.split('::')[0])
+    return exists ? <Emoji {...this.props} /> : <Text type='Body' style={{color: undefined, fontWeight: undefined}}>:{emoji}:</Text>
   }
 }
 
-const initialOpenToTag = {
-  '`': {Component: Text, props: {type: 'Body', style: codeSnippetStyle}},
-  '```': {Component: Text, props: {type: 'Body', style: codeSnippetBlockStyle}},
-  '*': {Component: Text, props: {type: 'BodySemibold', style: {color: undefined}}},
-  '_': {Component: Text, props: {type: 'Body', style: {fontStyle: 'italic', fontWeight: undefined, color: undefined}}},
-  '~': {Component: Text, props: {type: 'Body', style: {textDecoration: 'line-through', fontWeight: undefined, color: undefined}}},
-  ':': {Component: EmojiIfExists, props: {size: 16}},
-}
-
-const openToNextOpenToTag = {
-  '`': {},
-  '```': {},
-  '*': initialOpenToTag,
-  '_': initialOpenToTag,
-  '~': initialOpenToTag,
-  ':': {},
-}
-
-type TagMeta = {
-  componentInfo: {Component: ReactClass<*>, props: Object},
-  textSoFar: string,
-  elementsSoFar: List<React$Element<*> | string>,
-  openToTag: {[key: string]: TagInfo<Text> | TagInfo<EmojiIfExists>},
-  closingTag: ?string,
-}
-
-const initalTagMeta: TagMeta = {
-  componentInfo: plainStringTag,
-  textSoFar: '',
-  elementsSoFar: new List(),
-  openToTag: initialOpenToTag,
-  closingTag: null,
-}
-
-type TagStack = List<TagMeta>
-
-const marksRegex = new RegExp(Object.keys(openToClosePair).map(s => '^' + markToRegex[s]).join('|'))
-function matchWithMark (text: string): ?{matchingMark: string, restText: string} {
-  const m = text.match(marksRegex)
-  if (m && m[0]) {
-    const matchingMark = m[0]
-    return {matchingMark, restText: text.slice(matchingMark.length)}
+function createComponent (type, key, children) {
+  switch (type) {
+    case 'inline-code':
+      return <Text type='Body' key={key} style={codeSnippetStyle}>{children}</Text>
+    case 'code-block':
+      return <Text type='Body' key={key} style={codeSnippetBlockStyle}>{children}</Text>
+    case 'text':
+      return <Text type='Body' key={key} style={{color: undefined, fontWeight: undefined}}>{children}</Text>
```

Comment:
make sure to get my styles form https://github.com/keybase/client/pull/5388. we should make these style props consts so we don't get perf hits from making the objects over and over